### PR TITLE
Take DPR into account for image inversion

### LINKED ISSUE #87869: debugInvertOversizedImages does not take devicePixelRatio into account
## Steps to Reproduce

1. Run `flutter create bug`.
2. Change the default app to display an image and set `debugInvertOversizedImages` to true. For example like this:
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    debugInvertOversizedImages = true;

    return MaterialApp(
      home: Scaffold(
        body: Builder(builder: (context) {
          return Center(
            child: Image.network(
              'https://images.unsplash.com/photo-1609876322269-eac81abb26a7',
            ),
          );
        }),
      ),
    );
  }
}
```
3. See a warning like this:
```
════════ Exception caught by painting library ══════════════════════════════════
The following message was thrown while painting an image:
Image https://images.unsplash.com/photo-1609876322269-eac81abb26a7 has a display size of 384×256 but a decode size of 5184×3456, which uses an additional 92800KB.

Consider resizing the asset ahead of time, supplying a cacheWidth parameter of 384, a cacheHeight parameter of 256, or using a ResizeImage.

════════════════════════════════════════════════════════════════════════════════
```
4. Set the cacheWidth in `Image.network` to the number in the error message.
5. See low quality image.

**Expected results:** A larger image should be allowed before it is filpped and inverted. The exact size should be the `devicePixelRatio` multiplied with the size of the widget in logical pixels, i.e. the size in physical pixels.

**Actual results:** An error is thrown if the image resolution is higher than the widget size in logical pixels.

The problem has been discussed in https://github.com/flutter/flutter/issues/56332 as well, but that was after it was closed.

<details>
  <summary>Logs</summary>

```
[  +44 ms] executing: uname -m
[  +22 ms] Exit code 0 from: uname -m
[        ] x86_64
[   +4 ms] executing: [/home/hannes/fvm/versions/stable/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +4 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] f4abaa0735eba4dfd8f33f73363911d63931fe03
[        ] executing: [/home/hannes/fvm/versions/stable/] git tag --points-at f4abaa0735eba4dfd8f33f73363911d63931fe03
[  +10 ms] Exit code 0 from: git tag --points-at f4abaa0735eba4dfd8f33f73363911d63931fe03
[        ] 2.2.3
[   +4 ms] executing: [/home/hannes/fvm/versions/stable/] git rev-parse --abbrev-ref --symbolic @{u}
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/home/hannes/fvm/versions/stable/] git ls-remote --get-url origin
[   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +53 ms] executing: [/home/hannes/fvm/versions/stable/] git rev-parse --abbrev-ref HEAD
[   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +36 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +34 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb devices -l
[  +23 ms] List of devices attached
           192.168.1.216:5555     device product:OnePlus7Pro model:GM1913 device:OnePlus7Pro transport_id:1
[   +4 ms] /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell getprop
[ +140 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +64 ms] Skipping pub get: version match.
[  +70 ms] Generating /home/hannes/Documents/example/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +35 ms] ro.hardware = qcom
[  +21 ms] Initializing file store
[   +6 ms] Skipping target: gen_localizations
[   +4 ms] complete
[   +2 ms] Launching lib/main.dart on GM1913 in debug mode...
[   +3 ms] /home/hannes/fvm/versions/stable/bin/cache/dart-sdk/bin/dart --disable-dart-dev /home/hannes/fvm/versions/stable/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/hannes/fvm/versions/stable/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill /tmp/flutter_tools.UEUGAL/flutter_tool.FURDIB/app.dill --packages /home/hannes/Documents/example/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build/3c113a45063dc6628e68a4111abcacad.cache.dill.track.dill --enable-experiment=alternative-invalidation-strategy
[   +4 ms] executing: /home/hannes/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/hannes/Documents/example/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[   +4 ms] Exit code 0 from: /home/hannes/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/hannes/Documents/example/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1e
               A: android:compileSdkVersionCodename(0x01010573)="11" (Raw: "11")
               A: package="com.example.example" (Raw: "com.example.example")
               A: platformBuildVersionCode=(type 0x10)0x1e
               A: platformBuildVersionName=(type 0x10)0xb
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1e
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=16)
                 A: android:label(0x01010001)="example" (Raw: "example")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=21)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.example.MainActivity" (Raw: "com.example.example.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=35)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=45)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw: "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=49)
                     E: action (line=50)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=52)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=59)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +4 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell -x logcat -v time -t 1
[   +8 ms] <- compile package:example/main.dart
[ +271 ms] --------- beginning of main
           08-07 20:54:09.517 I/vendor.qti.bluetooth@1.0-ibs_handler(  954): ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
[  +12 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb version
[   +5 ms] Android Debug Bridge version 1.0.41
           Version 31.0.2-7242960
           Installed as /home/hannes/Android/Sdk/platform-tools/adb
[   +2 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb start-server
[   +5 ms] Building APK
[  +17 ms] Running Gradle task 'assembleDebug'...
[   +2 ms] Using gradle from /home/hannes/Documents/example/android/gradlew.
[  +19 ms] executing: /opt/android-studio/jre/bin/java -version
[  +84 ms] Exit code 0 from: /opt/android-studio/jre/bin/java -version
[        ] openjdk version "11.0.10" 2021-01-19
           OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
           OpenJDK 64-Bit Server VM (build 11.0.10+0-b96-7249189, mixed mode)
[   +2 ms] executing: [/home/hannes/Documents/example/android/] /home/hannes/Documents/example/android/gradlew -Pverbose=true -Ptarget-platform=android-arm64 -Ptarget=/home/hannes/Documents/example/lib/main.dart -Pdart-defines=RkxVVFRFUl9XRUJfQVVUT19ERVRFQ1Q9dHJ1ZQ== -Pdart-obfuscation=false -Ptrack-widget-creation=true -Ptree-shake-icons=false -Pfilesystem-scheme=org-dartlang-root assembleDebug
[ +705 ms] > Task :app:compileFlutterBuildDebug UP-TO-DATE
[        ] > Task :app:packLibsflutterBuildDebug UP-TO-DATE
[        ] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:preDebugBuild UP-TO-DATE
[        ] > Task :app:compileDebugAidl NO-SOURCE
[        ] > Task :app:compileDebugRenderscript NO-SOURCE
[        ] > Task :app:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :app:checkDebugAarMetadata UP-TO-DATE
[        ] > Task :app:cleanMergeDebugAssets
[        ] > Task :app:mergeDebugShaders UP-TO-DATE
[        ] > Task :app:compileDebugShaders NO-SOURCE
[        ] > Task :app:generateDebugAssets UP-TO-DATE
[        ] > Task :app:mergeDebugAssets
[ +184 ms] > Task :app:copyFlutterAssetsDebug
[        ] > Task :app:generateDebugResValues UP-TO-DATE
[        ] > Task :app:generateDebugResources UP-TO-DATE
[        ] > Task :app:mergeDebugResources UP-TO-DATE
[        ] > Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
[        ] > Task :app:extractDeepLinksDebug UP-TO-DATE
[        ] > Task :app:processDebugMainManifest UP-TO-DATE
[        ] > Task :app:processDebugManifest UP-TO-DATE
[        ] > Task :app:processDebugManifestForPackage UP-TO-DATE
[        ] > Task :app:processDebugResources UP-TO-DATE
[        ] > Task :app:compileDebugKotlin UP-TO-DATE
[        ] > Task :app:javaPreCompileDebug UP-TO-DATE
[        ] > Task :app:compileDebugJavaWithJavac UP-TO-DATE
[        ] > Task :app:compileDebugSources UP-TO-DATE
[        ] > Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
[        ] > Task :app:compressDebugAssets UP-TO-DATE
[        ] > Task :app:processDebugJavaRes NO-SOURCE
[        ] > Task :app:mergeDebugJavaResource UP-TO-DATE
[        ] > Task :app:checkDebugDuplicateClasses UP-TO-DATE
[        ] > Task :app:dexBuilderDebug UP-TO-DATE
[        ] > Task :app:desugarDebugFileDependencies UP-TO-DATE
[        ] > Task :app:mergeExtDexDebug UP-TO-DATE
[        ] > Task :app:mergeDexDebug UP-TO-DATE
[        ] > Task :app:mergeDebugJniLibFolders UP-TO-DATE
[        ] > Task :app:mergeDebugNativeLibs UP-TO-DATE
[   +1 ms] > Task :app:stripDebugDebugSymbols UP-TO-DATE
[        ] > Task :app:validateSigningDebug UP-TO-DATE
[        ] > Task :app:packageDebug UP-TO-DATE
[ +195 ms] > Task :app:assembleDebug
[        ] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
[        ] BUILD SUCCESSFUL in 1s
[        ] 32 actionable tasks: 4 executed, 28 up-to-date
[ +338 ms] Running Gradle task 'assembleDebug'... (completed in 1 534ms)
[ +166 ms] calculateSha: LocalDirectory: '/home/hannes/Documents/example/build/app/outputs/flutter-apk'/app.apk
[ +566 ms] ✓  Built build/app/outputs/flutter-apk/app-debug.apk.
[   +1 ms] executing: /home/hannes/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/hannes/Documents/example/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[   +5 ms] Exit code 0 from: /home/hannes/Android/Sdk/build-tools/30.0.3/aapt dump xmltree /home/hannes/Documents/example/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1e
               A: android:compileSdkVersionCodename(0x01010573)="11" (Raw: "11")
               A: package="com.example.example" (Raw: "com.example.example")
               A: platformBuildVersionCode=(type 0x10)0x1e
               A: platformBuildVersionName=(type 0x10)0xb
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1e
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=16)
                 A: android:label(0x01010001)="example" (Raw: "example")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=21)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.example.MainActivity" (Raw: "com.example.example.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=35)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=45)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw: "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=49)
                     E: action (line=50)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=52)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=59)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +1 ms] Stopping app 'app.apk' on GM1913.
[        ] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell am force-stop com.example.example
[ +270 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell pm list packages com.example.example
[ +113 ms] package:com.example.example
[   +2 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell cat /data/local/tmp/sky.com.example.example.sha1
[  +74 ms] f7b3ad6a9ed257b109044bf0fcffbdd0b965770b
[        ] Latest build already installed.
[        ] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell -x logcat -v time -t 1
[ +232 ms] --------- beginning of main
           08-07 20:54:12.807 D/MainServiceImplHandler: ( 3130): EVENT_STATE_INIT
[   +8 ms] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 shell am start -a android.intent.action.RUN -f 0x20000000 --ez enable-background-compilation true --ez enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true com.example.example/com.example.example.MainActivity
[ +109 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.example/.MainActivity (has extras) }
[        ] Waiting for observatory port to be available...
[ +380 ms] Observatory URL on device: http://127.0.0.1:43283/E7g0YRKA4Vo=/
[        ] executing: /home/hannes/Android/Sdk/platform-tools/adb -s 192.168.1.216:5555 forward tcp:0 tcp:43283
[   +7 ms] 35537
[        ] Forwarded host port 35537 to device port 43283 for Observatory
[+1603 ms] Caching compiled dill
[  +77 ms] Connecting to service protocol: http://127.0.0.1:35537/E7g0YRKA4Vo=/
[ +474 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:35537/E7g0YRKA4Vo=/.
[ +924 ms] DDS is listening at http://127.0.0.1:37191/e3t7XdUHCqo=/.
[  +64 ms] Successfully connected to service protocol: http://127.0.0.1:35537/E7g0YRKA4Vo=/
[  +30 ms] DevFS: Creating new filesystem on the device (null)
[  +65 ms] DevFS: Created new filesystem on the device (file:///data/user/0/com.example.example/code_cache/exampleSKRKCI/example/)
[   +2 ms] Updating assets
[  +60 ms] Syncing files to device GM1913...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +1 ms] <- recompile package:example/main.dart effaebd8-1c31-4477-89ac-a448c94f2754
[        ] <- effaebd8-1c31-4477-89ac-a448c94f2754
[  +61 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device GM1913... (completed in 64ms)
[        ] Synced 0.0MB.
[        ] <- accept
[  +30 ms] Connected to _flutterView/0xb40000742d5d2570.
[   +2 ms] Flutter run key commands.
[        ] r Hot reload. 🔥🔥🔥
[        ] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[        ] An Observatory debugger and profiler on GM1913 is available at: http://127.0.0.1:37191/e3t7XdUHCqo=/
[ +349 ms] The Flutter DevTools debugger and profiler on GM1913 is available at: http://127.0.0.1:9102?uri=http%3A%2F%2F127.0.0.1%3A37191%2Fe3t7XdUHCqo%3D%2F
```

```
Analyzing example...                                            
No issues found! (ran in 0.9s)
```

```
[✓] Flutter (Channel stable, 2.2.3, on Linux, locale sv_SE.UTF-8)
    • Flutter version 2.2.3 at /home/hannes/fvm/versions/stable
    • Framework revision f4abaa0735 (5 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/hannes/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /usr/bin/google-chrome-stable

[✓] Android Studio (version 4.2)
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Connected device (2 available)
    • GM1913 (mobile) • 192.168.1.216:5555 • android-arm64  • Android 11 (API 30)
    • Chrome (web)    • chrome             • web-javascript • Google Chrome 92.0.4515.131

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/87869

Also makes a small nullsafety adjustment.

/cc @hannnes1

### CODE REVIEW

File: dev/tracing_tests/README.md [hunk 1]
```diff
@@ -4,4 +4,4 @@ The tests in this folder must be run with `flutter test --enable-vmservice`,
 since they test that trace data is written to the timeline by connecting to
 the observatory.
 
-These tests will fail if run without this flag.
\ No newline at end of file
+These tests will fail if run without this flag.
```
Comment (goderbauer) [inline dev/tracing_tests/README.md:7]: I can't spot what has changed on this line...

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/tracing_tests/README.md (hunk 1 above)
Comment (dnfield) [reply dev/tracing_tests/README.md:7]: new line at EOF

Comment (goderbauer) [reply dev/tracing_tests/README.md:7]: Ah, I thought github's UI actually showed something to visualize that... Maybe I dreamt that.

Comment (dnfield) [reply dev/tracing_tests/README.md:7]: It puts a little icon at the end if it's missing, but I guess not when you add it

Comment (dnfield) [reply dev/tracing_tests/README.md:7]: I only noticed because I `cat`ed the file and it didn't print nicely

**Pull request merged**